crates/proto/src/domain/notes.rs

```diff
@@ -15,9 +15,16 @@ impl TryFrom<crate::generated::note::NoteMetadata> for NoteMetadata {
             .try_into()?;
         let note_type = NoteType::try_from(value.note_type as u64)?;
         let tag = NoteTag::from(value.tag);
+
+        // TODO: Conversion/helper functions should be provided for these conversions
+        let execution_hint_tag = (value.execution_hint & 0xFF) as u8;
+        let execution_hint_payload = ((value.execution_hint >> 8) & 0xFFFFFF) as u32;
+        let execution_hint =
+            NoteExecutionHint::from_parts(execution_hint_tag, execution_hint_payload)?;
```

Comment:
Something to consider as part of the discussion in #435

Reply:
I think instead of doing this, we should actually use `NoteExecutionHint::try_from()` here (I believe this is now available in `miden-base`).